### DISCUSSION TITLE
Unable to download packages due to error in a subprocess.

### DISCUSSION THREAD

**Discussion by samiamrg7** [LykosAI/StabilityMatrix#433] [Q&A]
I'm trying to install a package using Stability matrix, and I keep getting errors related to pip and some "subprocess." It recommends that I "use pep517," but when I try, it returns an error and I can't figure out how to make that work.

I have tried to install Stable Diffusion Webui  and ComfyUI, but both encountered similar errors. 
I'm on windows.
Here is a portion of the error code.

Collecting GitPython==3.1.32

  Using cached GitPython-3.1.32-py3-none-any.whl.metadata (10.0 kB)

Collecting Pillow==9.5.0

  Using cached Pillow-9.5.0-cp310-cp310-win_amd64.whl (2.5 MB)

Collecting accelerate==0.21.0

  Using cached accelerate-0.21.0-py3-none-any.whl.metadata (17 kB)

Collecting basicsr==1.4.2

  Using cached basicsr-1.4.2.tar.gz (172 kB)

  Preparing metadata (setup.py): started

  Preparing metadata (setup.py): still running...

  Preparing metadata (setup.py): finished with status 'error'

  error: subprocess-exited-with-error

  python setup.py egg_info did not run successfully.

  exit code: 1

  [44 lines of output]

  D:\Documents D\A1111 Web UI Autoinstaller\StabilityMatrix-win-x64\Data\Packages\stable-diffusion-webui\venv\lib\site-packages\setuptools_init_.py:80: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated.

  !!

          **

          Requirements should be satisfied by a PEP 517 installer.

          If you are using pip, you can try pip install --use-pep517.

          **

  !!

    dist.fetch_build_eggs(dist.setup_requires)

  Traceback (most recent call last):

    File "<string>", line 2, in <module>

    File "<pip-setuptools-caller>", line 34, in <module>

    File "C:\Users(file name redacted\AppData\Local\Temp\pip-install-n23h_l2i\basicsr_b75c4f1669d0446f9eaa8e45180ef1ea\setup.py", line 147, in <module> 
    
    
Here is some more text from the error that appears to diagnose the problem, but I don't understand what it means or what to do about it. 

FileNotFoundError: [Errno 2] No such file or directory: 'C:\Users\(file name redacted\AppData\Local\Temp\pip-install-n23h_l2i\basicsr_b75c4f1669d0446f9eaa8e45180ef1ea\.eggs\torch-2.2.0-py3.10-win-amd64.egg\torch\include\ATen\ops\_fake_quantize_per_tensor_affine_cachemask_tensor_qparams_compositeexplicitautograd_dispatch.h'

  [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.

error: metadata-generation-failed

Encountered error while generating package metadata.

See above for output.

note: This is an issue with the package mentioned above, not pip.

hint: See above for details.

Error: StabilityMatrix.Core.Exceptions.ProcessException: pip install failed with code 1: 'Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cu121/r/nCollecting torch==2.1.2\r\n  Using cached https://download.pytorch.org/whl/cu121/torch-2.1.2%2Bcu121-cp310-cp310-win_amd64.whl (2473.9 MB)\r\nCollecting torchvision==0.16.2\r\n  Downloading https://download.pytorch.org/whl/cu121/torchvision-0.16.2%2Bcu121-cp310-cp310-win_amd64.whl (5.6 MB)\r\n
5.6/5.6 MB 3.1 MB/s eta 0:00:00\r\nCollecting GitPython==3.1.32\r\n  Using cached GitPython-3.1.32-py3-none-any.whl.metadata (10.0 kB)\r\nCollecting Pillow==9.5.0\r\n  Using cached Pillow-9.5.0-cp310-cp310-win_amd64.whl (2.5 MB)\r\nCollecting accelerate==0.21.0\r\n  Using cached accelerate-0.21.0-py3-none-any.whl.metadata (17 kB)\r\nCollecting basicsr==1.4.2\r\n  Using cached basicsr-1.4.2.tar.gz (172 kB)\r\n  Preparing metadata (setup.py): started\r\n  Preparing metadata (setup.py): still running...\r\n  Preparing metadata (setup.py): finished with status 'error'\r\n  error: subprocess-exited-with-error\r\n  \r\n  python setup.py egg_info did not run successfully.\r\n  exit code: 1\r\n  \r\n  [44 lines of output]\r\n  D:\Documents D\A1111 Web UI Autoinstaller\StabilityMatrix-win-x64\Data\Packages\stable-diffusion-webui\venv\lib\site-packages\setuptools_init_.py:80: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated.\r\n  !!\r\n  \r\n          **\r\n          Requirements should be satisfied by a PEP 517 installer.\r\n          If you are using pip, you can try pip install --use-pep517.\r\n

**Comment by truefriend-cz** [comment]
I didn't want to ramble here but I have the same error and I'm still waiting for a response from the developer with a solution, but obviously I'd sooner give birth to a pink elephant than wait for a fix or a response.

**Comment by ionite34** [comment]
Hi, which package did you install in SM exactly, "Stable Diffusion WebUI" or "ComfyUI"? And which version option did you select, if default it should be the latest release for WebUI and latest main branch for ComfyUI.

It seems a dependency install encountered an issue "basicsr" but neither of our WebUI nor ComfyUI's installs have that dependency, so I'm a bit confused as to which install process this is?

**Comment by lakoske** [comment]
Hello. I have same problem. when installing Stable Diffusion WebUI (A1111) I get an error

note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Error: StabilityMatrix.Core.Exceptions.ProcessException: pip install failed with code 1: 'Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cu121\r\nCollecting torch==2.1.2\r\n  Downloading https://download.pytorch.org/whl/cu121/torch-2.1.2%2Bcu121-cp310-cp310-win_amd64.whl (2473.9 MB)\r\n     

tnx
